src/Archives/BitStream.cpp

```diff
@@ -93,7 +93,7 @@ namespace Archives
 		return m_ReadBitIndex >= m_BufferBitSize;
 	}
 
-	int BitStream::GetBitReadPos() const
+	std::size_t BitStream::GetBitReadPos() const
```

Comment:
That's not a terrible idea. Thought for 64-bit builds, it's still the same issue, since an `std::size_t` is likely going to be the same size as a `uint64_t` there.

I think in practice, we could just check and throw an exception. If we stuck with `std::size_t`, this means on 32-bit, max array size is 4GB, but max bit stream length is 512MB (4GB / 8). That's still plenty large. On 64-bit, max array size is 2^64 bytes, with max bit stream length 2^64 / 8 = 2^61 bytes.

I would be happy with a constructor check that ensures buffer size is capped at `std::numeric_limits<std::size_t>::max() / 8`, and stuck with using `std::size_t`.

Note: The above [`numeric_limits`](https://en.cppreference.com/w/cpp/types/numeric_limits/max) template is from `#include <limits>`

Reply:
I would have probably used SIZE_MAX since I wasn't aware of the std::numeric_limits<std::size_t>::max() syntax. Do you prefer one over the other for any reason?

-Brett

Reply:
Actually, looking at this a little closer, it doesn't make sense to compare with a max size since it will wrap around and be smaller than max size before comparison. what about:

```
// Check if storing bufferSize in bits wraps past largest value of a std::size_t
if (bufferSize << 3 < bufferSize) {
	throw std::runtime_error("BitStream cannot support a buffer size of " + std::to_string(bufferSize));
}
```